app/src/main/java/org/mozilla/focus/components/ComponentToggleService.java

```diff
@@ -0,0 +1,257 @@
+        final ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (ComponentToggleService.class.getName().equals(service.service.getClassName())) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private void startToForeground() {
+        //  The notification channel id should not be necessary since this service should only be enabled for API 21~23
+        final NotificationCompat.Builder builder = new NotificationCompat.Builder(
+                getApplicationContext(),
+                NOTIFICATION_CHANNEL);
+
+        final Notification notification = builder
+                .setContentTitle(getString(R.string.setting_default_browser_notification_title))
+                .setContentText(getString(R.string.setting_default_browser_notification_text))
+                .setBadgeIconType(R.drawable.ic_notification)
+                .setSmallIcon(R.mipmap.ic_launcher)
+                .setAutoCancel(false)
+                .setOngoing(true)
+                .setContentIntent(buildIntent())
+                .setPriority(NotificationCompat.PRIORITY_HIGH)
+                .build();
+
+        startForeground(FG_NOTIFICATION_ID, notification);
+    }
+
+    private void removeFromForeground() {
+        // to post a new notification so people can go to SettingsActivity easily
+        // this notification will be removed by SettingsActivity if it is in foreground
+        final NotificationCompat.Builder builder = new NotificationCompat.Builder(
+                getApplicationContext(),
+                NOTIFICATION_CHANNEL);
+        final Notification notification = builder
+                .setContentTitle(getString(R.string.setting_default_browser_notification_clickable_text))
+                .setBadgeIconType(R.drawable.ic_notification)
+                .setSmallIcon(R.mipmap.ic_launcher)
+                .setAutoCancel(true)
+                .setContentIntent(buildIntent())
+                .build();
+
+        NotificationManagerCompat.from(getApplicationContext())
+                .notify(NOTIFICATION_ID, notification);
+
+        if (timer != null) {
+            timer.cancel();
+            timer = null;
+        }
+
+        stopSelf();
+    }
+
+    private PendingIntent buildIntent() {
+        return PendingIntent.getActivity(getApplicationContext(),
+                INTENT_REQ_CODE,
+                new Intent(getApplicationContext(), SettingsActivity.class),
+                PendingIntent.FLAG_UPDATE_CURRENT);
+    }
+
+    // A TimerTask to stop service if it runs too long
+    private final static class BombTask extends TimerTask {
+        // How long this Service will stop itself
+        private static final long TIMEOUT = 30000; // 30 seconds
+
+        final ComponentToggleService service;
```

Comment:
Maybe use a weakReference?

Reply:
The task itself will be canceled once the service be destroyed. To use WeakReference here won't hurt anything, so I updated it as your suggestion.